src/Microsoft.Data.Analysis/DataFrame.IO.cs

```diff
@@ -500,6 +525,19 @@ public TextReader GetTextReader()
                                 continue;
                             }
 
+                            if (t == typeof(string))
+                            {
+                                // TODO why doesn't Contains(char) work? 
+                                bool needsQuotes = ((string)cell).Contains(separator.ToString()) || ((string)cell).Contains("\n");
```

Comment:
I'm kinda stumped on this, confused why I have to convert separator to a string when this supposedly exists https://docs.microsoft.com/en-us/dotnet/api/system.string.contains?view=net-6.0